crates/miden-testing/src/tx_context/context.rs

```diff
@@ -136,7 +129,7 @@ impl TransactionContext {
     ///
     /// For more information, see the docs for [TransactionContext::execute_code_with_assembler()].
     pub fn execute_code(&self, code: &str) -> Result<Process, ExecutionError> {
-        let assembler = TransactionKernel::with_kernel_library();
+        let assembler = TransactionKernel::with_kernel_library(self.source_manager.clone());
         self.execute_code_with_assembler(code, assembler)
```

Comment:
I think the changes to `execute_code_with_assembler` should work now, but trying the test I posted in the other comment does not yet include the source snippet.

I think this is because the `MockHost` does not yet return the correct source to the processor in its host implementation - there is a `TODO: SourceManager` that I think needs to be resolved. Essentially, by making the `MockHost` own the source manager as well from which it can load source files and respond to source requests by the processor.

(Side note, there are overall two `TODO: SourceManager` left which would be good to address in this PR).